System/Metrics/Prometheus/Gauge.hs

```diff
@@ -25,9 +26,15 @@ newtype Gauge = C { unC :: Atomic.Atomic }
 new :: IO Gauge
 new = C `fmap` Atomic.new 0
 
+-- | Get the current value of the gauge as a 'Double'.
+--
+-- > read = fmap fromIntegral . readInt
+read :: Gauge -> IO Double
+read = fmap fromIntegral . readInt
+
 -- | Get the current value of the gauge.
-read :: Gauge -> IO Int64
-read = Atomic.read . unC
```

Comment:
My impression was that Int64s were being replaced with Doubles, is this function used? Sorry if I missed something and Int gauges are being used after all.

Reply:
My reasoning was that we would provide `read :: Gauge -> IO Double` for convenience, since that is what one must ultimately use to register the gauge to the metrics store, and `readInt :: Gauge -> IO Int64` for flexibility, since that is the underlying representation.

Strictly speaking, I do use `readInt` in one of the tests to test equality at `Int64` instead of at `Double`, but that's not a very compelling use.